lib/spack/spack/__init__.py

```diff
@@ -188,7 +188,7 @@
 __all__ += spack.util.executable.__all__
 
 # User's editor from the environment
-editor = Executable(os.environ.get("EDITOR", "vi"))
+editor = Executable(os.environ.get("EDITOR", "vim"))
```

Comment:
Let me back out this change and make it in a different PR, just to keep things simple.